99 search space stats

rohanpaul opened the pull request.

rohanpaul committed:
phrase class changes

rohanpaul committed:
string property added for the statistics

rohanpaul committed:
testing the search space size

rohanpaul committed:
added

rohanpaul committed:
minor

rohanpaul committed:
adcg_demo

rohanpaul committed:
test

tmhoward committed:
updated grounding set from_xml for container_type

rohanpaul committed:
search space statistics

rohanpaul committed:
adcg_test minor mods

rohanpaul committed:
test process

jarkinUR commented on src/symbol/grounding_set.cc:
```diff
@@ -308,6 +310,8 @@ from_xml( xmlNodePtr root ){
           _groundings.push_back( new Constraint( l1 ) );
         } else if ( xmlStrcmp( l1->name, ( const xmlChar* )( "container" ) ) == 0 ){
           _groundings.push_back( new Container( l1 ) );
+        } else if ( xmlStrcmp( l1->name, ( const xmlChar* )( "container_type" ) ) == 0 ){
```
Nice catch!

jarkinUR replied:
Hmmm, is this being included twice? Looks like lines 287-288 include the "rule_container_type" already. Lines 325-326 include "container_type".

arkinj commented:
lgtm*
edit: should check the grounding_set.cc 'from_xml' function to make sure 'rule_container_type' and 'container_type' are not being loaded twice.

Commit:
storing solutions to xml; made grammar an optional arg

Commit:
Removed duplicate loading of container_type and rule_container_type

Note that this would likely not produce any issues since it is inside of a
cascading if/else statement; the second version of the duplicate would simply
never get called.

rohanpaul committed:
test process

rohanpaul committed:
added

rohanpaul committed:
adcg

rohanpaul committed:
added

rohanpaul committed:
adcg test

rohanpaul committed:
stats

rohanpaul committed:
added

rohanpaul committed:
added
